src/common/attribute.rs

```diff
@@ -0,0 +1,626 @@
+		win_gravity: Option<WinGravity>, backing_store: Option<BackingStore>,
+		backing_planes: Option<u32>, backing_pixel: Option<Pixel>, override_redirect: Option<bool>,
+		save_under: Option<bool>, event_mask: Option<EventMask>,
+		do_not_propagate_mask: Option<DeviceEventMask>,
+		colormap: Option<CopyableFromParent<Colormap>>,
+		cursor_appearance: Option<Option<CursorAppearance>>,
+	) -> Self {
+		let mut x11_size = 0;
+		let mut mask = AttributeMask::empty();
+
+		if let Some(background_pixmap) = &background_pixmap {
+			x11_size += background_pixmap.x11_size();
+			mask |= AttributeMask::BACKGROUND_PIXMAP;
+		}
+		if let Some(background_pixel) = &background_pixel {
+			x11_size += background_pixel.x11_size();
+			mask |= AttributeMask::BACKGROUND_PIXEL;
+		}
+
+		if let Some(border_pixmap) = &border_pixmap {
+			x11_size += border_pixmap.x11_size();
+			mask |= AttributeMask::BORDER_PIXMAP;
+		}
+		if let Some(border_pixel) = &border_pixel {
+			x11_size += border_pixel.x11_size();
+			mask |= AttributeMask::BORDER_PIXEL;
+		}
+
+		if let Some(bit_gravity) = &bit_gravity {
+			x11_size += bit_gravity.x11_size();
+			mask |= AttributeMask::BIT_GRAVITY;
+		}
+		if let Some(win_gravity) = &win_gravity {
+			x11_size += win_gravity.x11_size();
+			mask |= AttributeMask::WIN_GRAVITY;
+		}
+
+		if let Some(backing_store) = &backing_store {
+			x11_size += backing_store.x11_size();
+			mask |= AttributeMask::BACKING_STORE;
+		}
+		if let Some(backing_planes) = &backing_planes {
+			x11_size += backing_planes.x11_size();
+			mask |= AttributeMask::BACKING_PLANES;
+		}
+		if let Some(backing_pixel) = &backing_pixel {
+			x11_size += backing_pixel.x11_size();
+			mask |= AttributeMask::BACKING_PIXEL;
+		}
+
+		if let Some(override_redirect) = &override_redirect {
+			x11_size += override_redirect.x11_size();
+			mask |= AttributeMask::OVERRIDE_REDIRECT;
+		}
+		if let Some(save_under) = &save_under {
+			x11_size += save_under.x11_size();
+			mask |= AttributeMask::SAVE_UNDER;
+		}
+
+		if let Some(event_mask) = &event_mask {
+			x11_size += event_mask.x11_size();
+			mask |= AttributeMask::EVENT_MASK;
+		}
+		if let Some(do_not_propagate_mask) = &do_not_propagate_mask {
+			x11_size += do_not_propagate_mask.x11_size();
+			mask |= AttributeMask::DO_NOT_PROPAGATE_MASK;
+		}
+
+		if let Some(colormap) = &colormap {
+			x11_size += colormap.x11_size();
+			mask |= AttributeMask::COLORMAP;
+		}
+
+		if let Some(cursor_appearance) = &cursor_appearance {
+			x11_size += cursor_appearance.x11_size();
+			mask |= AttributeMask::CURSOR_APPEARANCE;
+		}
+
+		Self {
+			x11_size,
+			mask,
+
+			background_pixmap,
+			background_pixel,
+
+			border_pixmap,
+			border_pixel,
+
+			bit_gravity,
+			win_gravity,
+
+			backing_store,
+			backing_planes,
+			backing_pixel,
+
+			// These booleans are converted into our [`Bool`] type so that they can easily be
+			// written as four bytes.
+			override_redirect: override_redirect.map(std::convert::Into::into),
```

Comment:
Either this should use `Into::into`, or the other uses of `Into::into` should use `std::convert::Into::into`.

This is just a readability issue, exact same function it's referring to.